modules/core/src/effects/post-process-effect.ts

```diff
@@ -35,19 +35,16 @@ export default class PostProcessEffect<ShaderPassT extends ShaderPass> implement
 
     for (let index = 0; index < this.passes.length; index++) {
       const isLastPass = index === this.passes.length - 1;
-      const renderToTarget = target !== undefined && isLastPass;
-      if (renderToTarget) {
+      if (target !== undefined && isLastPass) {
```

Comment:
When rendering to screen, target is `null`. This results in `outputBuffer` being set to `null`. If we then switch buffers below, the function will return `null`.

[DeckRenderer will then](https://github.com/visgl/deck.gl/blob/master/modules/core/src/lib/deck-renderer.ts#L166) pass this `null` as the `inputBuffer` into the next post process effect, causing a crash

You can repro in `layer-browser/src/app.tsx' by doing the following:

```ts
_getEffects() {
  // TODO
  const {shadow, postProcessing} = this.state.settings;

  return [
    shadow ? GLOBAL_LIGHTING_WITH_SHADOW : GLOBAL_LIGHTING,
    new PostProcessEffect(triangleBlur, {strength: 1}),
    new PostProcessEffect(ink, {strength: 1})
  ].filter(Boolean);
}
```

Reply:
I see what's happening there. The original logic was broken by the v9 refactor but your fix was also incorrect.

The second-to-last PostProcessEffect should:
- keep rendering to swap buffer and return the last fbo written into (the current master writes to screen them return the output of the previous pass)

The second-to-last PostProcessEffect should:
- keep rendering to swap buffer until the last pass, which renders into the final target (screen)

I've added some comments to my fix.

Reply:
Before (lost a pass):

![Screen Shot 2024-03-15 at 12 17 59 PM](https://github.com/visgl/deck.gl/assets/2059298/21659459-ca72-42ee-8aa7-1694a77f4d4d)

After:

![Screen Shot 2024-03-15 at 12 18 21 PM](https://github.com/visgl/deck.gl/assets/2059298/96646856-75de-4136-9009-b2c9923d1f97)

Reply:
Yes, makes sense. It could be good to add a Typescript function overloads to make it clear that the function can in fact return `null` and that the caller can only rely on a `Framebuffer` being returned when `target` is `undefined`. This would have caught the bug in `deck-renderer`